core/src/main/java/org/elasticsearch/common/util/concurrent/EsExecutors.java

```diff
@@ -60,16 +61,11 @@ public static PrioritizedEsThreadPoolExecutor newSinglePrioritizing(String name,
 
     public static EsThreadPoolExecutor newScaling(String name, int min, int max, long keepAliveTime, TimeUnit unit, ThreadFactory threadFactory, ThreadContext contextHolder) {
         ExecutorScalingQueue<Runnable> queue = new ExecutorScalingQueue<>();
-        // we force the execution, since we might run into concurrency issues in offer for ScalingBlockingQueue
```

Comment:
can we add a comment about using the `ForceQueuePolicy` capturing our research? will be a shame to have to do it again :)